### PR TITLE
[wave2-961] validate supervisor email to ensure ucalgary.ca domain

### PR DESCRIPTION
**Pull request by mhughes2012**

### CODE REVIEW

**Comment by mhughes2012** [comment]
Resolved conflict and pushed.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 1]
```diff
@@ -94,6 +100,24 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
     }
 }
 
+function fgs_email_domain_validation($form, &$form_state) {
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: Code Comment please!

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 2]
```diff
@@ -94,6 +100,24 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
...
+
+    // valid domains (one so far)
+    $UCDomains = array('ucalgary.ca');
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: better to use drupal variable for 'ucalgary.ca', so we can update it without code deploy.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 3]
```diff
@@ -94,6 +100,24 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
...
+
+    // Get domain
+    $email = $form_state['values']['field_gbl_supervisor_form_email'][LANGUAGE_NONE][0]['email'];
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: I guess should check first if `$form_state['values']['field_gbl_supervisor_form_email']` is set

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 4]
```diff
@@ -94,6 +100,24 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
...
+    $splitIndex = strpos($email, '@')+1;
+    $emailLength = strlen($email);
+    $domain = substr($email, $splitIndex, $emailLength);
+
+    // Error if not legitimate domain
+    if ( !in_array($domain, $UCDomains) ) {
+      form_set_error('mail', 'Email address must be a UCalgary email address.');
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: missing t() function

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 5]
```diff
@@ -94,6 +98,24 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
...
+
+    // Get domain
+    $email = $form_state['values']['field_gbl_supervisor_form_email'][LANGUAGE_NONE][0]['email'];
+    $splitIndex = strpos($email, '@') + 1;
+    $emailLength = strlen($email);
+    $domain = substr($email, $splitIndex, $emailLength);
+
+    // Error if not legitimate domain
+    if (!in_array($domain, $UCDomains)) {
+        form_set_error('mail', 'Email address must be a UCalgary email address.');
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: missing t()

**Comment by mhughes2012** [comment]
Since email is required if the form is showing or not, validating the email domain will still be called.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 2 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: thanks @balrajb, updated to a drupal variable.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 3 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: using an isset now.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 4 above)
Comment (mhughes2012) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: added t() function

**Pull request merged**